### PR TITLE
TM-4819-BE-Org-Stat

### PR DESCRIPTION
**Pull request by cagbakwu**
[Ticket](https://metaphase.atlassian.net/browse/TM-4918)
[Mock](https://github.com/MetaPhase-Consulting/mock-fsbid/pull/432)
[FE](https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/2848)

### CODE REVIEW

**Comment by cagbakwu** [comment]
Updating Filters...

File: talentmap_api/fsbid/services/org_stats.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        'results': list(map(list_org_stats_map, data.get('QRY_LSTORGSTATS_REF'))),
+        'bureau_summary': list(map(list_bureau_stats_map, data.get('QRY_LSTBUREAUSTATS_REF'))),
+    }
+
+def get_org_stats_filters(jwt_token):
+    '''
+    Gets Filters for Org Stats Page
+    '''
+    args = {
+        'proc_name': 'qry_lstfsbidSearch',
```
Comment (M00staff) [inline talentmap_api/fsbid/services/org_stats.py:84]: these filters are the same filters used for `publishablePositionFilters` ... is that by design?

File: talentmap_api/fsbid/services/org_stats.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    return services.send_post_back_office(
+        **args
+    )
+def org_stats_req_mapping(request):
+    return {
+        'pv_api_version_i': '',
+        'pv_ad_id_i': '',
+        'i_cycle_id': '',
+        'i_bureau_id': '',
+        'i_org_code': '',
```
Comment (M00staff) [inline talentmap_api/fsbid/services/org_stats.py:27]: passing blank values here means that the filters wont do anything - is that by design?

**Comment by M00staff** [comment]
test these branches out on dev and see if its working

File: talentmap_api/fsbid/services/org_stats.py (hunk 1 above)
Comment (cagbakwu) [reply talentmap_api/fsbid/services/org_stats.py:84]: Yea, they are grabbing from the same data set.

File: talentmap_api/fsbid/services/org_stats.py (hunk 2 above)
Comment (cagbakwu) [reply talentmap_api/fsbid/services/org_stats.py:27]: @M00staff ah sorry, forgot to push those changes up. Pushing now.

File: talentmap_api/fsbid/views/org_stats.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        Get All 
+        '''
+        result = services.get_org_stats(request.query_params, request.META['HTTP_JWT'])
+        if result is None:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        return Response(result)
+
+class FSBidOrgStatsFiltersView(APIView):
+    # perms TBD
```
Comment (M00staff) [inline talentmap_api/fsbid/views/org_stats.py:28]: did you get this cleared up?

File: talentmap_api/fsbid/services/org_stats.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        "proc_name": 'qry_lstorgstats',
+        "package_name": 'PKG_WEBAPI_WRAP_SPRINT101',
+        "request_mapping_function": org_stats_req_mapping,
+        "response_mapping_function": org_stats_res_mapping,
+        "jwt_token": jwt_token,
+        "request_body": query,
+    }
+    return services.send_post_back_office(
+        **args
+    )
```
Comment (M00staff) [inline talentmap_api/fsbid/services/org_stats.py:20]: nitpick - add a blank line under here

File: talentmap_api/fsbid/services/org_stats.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+def org_stats_req_mapping(request):
+    return {
+        'pv_api_version_i': '',
+        'pv_ad_id_i': '',
+        'i_cycle_id': request.get('cycles') or 0,
+        'i_bureau_cd': request.get('bureaus') or '',
+        'i_org_code': request.get('orgs') or '',
+    }
```
Comment (M00staff) [inline talentmap_api/fsbid/services/org_stats.py:28]: nitpick - add a blank line under here

**Review by M00staff — APPROVED** [review]
double checked this in dev and it looks great and is working - great work! 🙌

two nitpicks and one open question - update & answer please, but gonna go ahead and mark this one as approved

**Pull request merged**